src/excited_workflow/carbon_tracker.py

```diff
@@ -0,0 +1,371 @@
+    return rmse_da
+
+
+def create_scatterplots(
+    predictions: list[xr.Dataset],
+    targets: list[xr.Dataset],
+    y_key: str,
+    target_units: str,
+    output_dir: Path,
+    text: str,
+) -> str:
+    """Create scatterplots of prediction vs. target.
+
+    Args:
+        predictions: list of predictions datasets per group.
+        targets: list of target datasets per group.
+        y_key: name of target variable.
+        target_units: units for target variable.
+        output_dir: Path to output directory
+        text: text for markdown file
+    """
+    for idx, (target, prediction) in enumerate(zip(targets, predictions, strict=True)):
+        plt.figure(figsize=(5, 5))
+        plt.scatter(prediction["prediction_label"], target[y_key], s=20)
+        plt.axline((0, 0), slope=1, color="black")
+        plt.xlabel(f"{target[y_key].name} ({target_units})")
+        plt.ylabel(f"{prediction['prediction_label'].name} ({target_units})")
+        plt.tight_layout()
+        plt.savefig(output_dir / f"scatter{idx}.png")
+        plt.close()
+        text += f"**Scatter plot {idx}**\n ![image]({output_dir}/scatter{idx}.png)\n\n"
+
+    return text
+
+
+def make_full_plot(
+    predictions: list[xr.Dataset],
+    targets: list[xr.Dataset],
+    y_key: str,
+    target_units: str,
+    output_dir: Path,
+    text: str,
+) -> str:
+    """Make scatterplot for all groups.
+
+    Args:
+        predictions: list of predictions datasets per group.
+        targets: list of target datasets per group.
+        y_key: name of target variable.
+        target_units: units for target variable.
+        output_dir: Path to output directory
+        text: text for markdown file
+    """
+    fig = plt.figure(figsize=(6, 6))
+    ax = plt.axes()
+    for target, prediction in zip(targets, predictions, strict=True):
+        ax.scatter(target[y_key], prediction["prediction_label"], s=12, alpha=0.5)
+        ax.axline((0, 0), slope=1, color="black")
+    ax.set_xlabel(f"{target[y_key].name} ({target_units})")
+    ax.set_ylabel(f"{prediction['prediction_label'].name} ({target_units})")
+    fig.tight_layout()
+    fig.savefig(output_dir / "all_scatter.png")
+    fig.clf()
+    text += f"**Scatter plot all**\n ![image]({output_dir}/all_scatter.png)\n\n"
+
+    return text
+
+
+def create_rmseplots(rmses: list[xr.DataArray], output_dir: Path, text: str) -> str:
+    """Create map plot for rmse.
+
+    Args:
+        rmses: list of rmse dataarrays
+        output_dir: Path to output directory
+        text: text for markdown file
+    """
+    for idx, rmse in enumerate(rmses):
+        rmse.to_netcdf(output_dir / f"rmse{idx}.nc")
+        plt.figure(figsize=(5, 3))
+        rmse.plot()
+        plt.tight_layout()
+        plt.savefig(output_dir / f"rmseplot{idx}.png")
+        plt.close()
+        text += f"**RMSE map {idx}**\n ![image]({output_dir}/rmseplot{idx}.png)\n\n"
+
+    return text
+
+
+def create_markdown_file(
```

Comment:
I think it could be nice to add some explanatory text in the markdown file for people who aren't fully familiar with the workflow.